### PR TITLE
[TTAHUB-981] Fix OE Objectives Saving and Read Only View

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This ticket fixes two issues:

1. Error when saving objectives on a other-entity report.
2. Display saved goals in a read only view.

**Update 0907:** To return to the report and see the objectives in read only mode you will need to comment out validation for the fields we currently do NOT save (roles, topics, and resources) - GoalValidator.js -unfinishedObjectives().

## How to test

1. Create an other entity report.
2. Saving objectives on the report should no longer error.
3. Once the objectives are saved you should be redirected to a read only view.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-981

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
I ran into a white screen of death

Steps to reproduce:
1) Create a report and add a couple of objectives. I had two other entity recipients and I added resources to one and not to the other.
2) Leave report (go back to activity reports landing page)
3) Go to edit report, white screen of death

I tried twice, the second time clearing local storage before trying to edit.

File: src/services/activityReports.js [hunk 1]
```diff
@@ -382,6 +382,16 @@ export async function activityReportAndRecipientsById(activityReportId) {
       {
         model: Objective,
         as: 'objectivesWithoutGoals',
+        include: [
```
Comment (thewatermethod) [inline src/services/activityReports.js:385]: Will test shortly, but shouldn't this also include files & resources?

**Comment by thewatermethod** [comment]
Found a possible bug
- Create a new other entity report
- Add an objective, save, you are in supporting attachments
- Return to the /activity-reports landing page
- Return to the activity report, go to the goals & objectives tab, the objective is empty (nothing in the title, tta provided, etc.) I would expect to see no topics and that stuff, which is fine, but I would expect to see a title. The objectivesWithoutGoals array returned from the API is empty.

**Comment by AdamAdHocTeam** [comment]
> * jective is empty (nothing in the title, tta provided, etc.) I would expect to see no topics and that stuff, which is fine, but I would expect to see a title. The objectivesWithoutGoals array returned from the API is empty.

I cannot reproduce this issue. I might be doing something wrong. Also please try clearing you local storage. I had a weird issue yesterday were I think it was using the local storage data.

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**